### PR TITLE
[CN-583] Add Cache CR

### PR DESCRIPTION
**Pull request by SeriyBg**
Add Cache CR

### CODE REVIEW

File: controllers/hazelcast/cache_controller.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CacheReconciler reconciles a Cache object
+type CacheReconciler struct {
+	client.Client
+	Log              logr.Logger
+	Scheme           *runtime.Scheme
+	phoneHomeTrigger chan struct{}
+}
```
Comment (cagric0) [inline controllers/hazelcast/cache_controller.go:25]: Just a reminder, after [Refactor HZ client PR](https://github.com/hazelcast/hazelcast-platform-operator/pull/425) is merged, we should update _CacheReconciler_ by adding _ClientRegistry_.

File: controllers/hazelcast/multimap_controller.go [hunk 2]
```diff
@@ -98,6 +98,7 @@ func fillMultiMapConfigInput(multiMapInput *codecTypes.MultiMapConfig, mm *hazel
 
 	mms := mm.Spec
 	multiMapInput.BackupCount = *mms.BackupCount
+	multiMapInput.AsyncBackupCount = *mms.AsyncBackupCount
 	multiMapInput.Binary = mms.Binary
 	multiMapInput.CollectionType = string(mms.CollectionType)
```
Comment (cagric0) [inline controllers/hazelcast/multimap_controller.go:103]: I think, _AsyncBackupCount_ field should be added to [fillAddMapConfigInput](https://github.com/hazelcast/hazelcast-platform-operator/blob/CN-583-cache-cr/controllers/hazelcast/map_controller.go#L298) same as here

**Review by cagric0 — CHANGES_REQUESTED** [review]
Thanks for additional refactoring, I've left a few comment. Other than that LGTM!

File: api/v1alpha1/cache_types.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+// CacheStatus defines the observed state of Cache
+type CacheStatus struct {
+	DataStructureStatus `json:",inline"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.state",description="Current state of the Queue Config"
```
Comment (dzeromski-hazelcast) [inline api/v1alpha1/cache_types.go:31]: Should it be queue or cache in the description?

**Review by dzeromski-hazelcast — APPROVED** [review]

**Review by cagric0 — APPROVED** [review]

**Pull request merged**